typeclasses/npcshop/npcshop.py

```diff
@@ -0,0 +1,269 @@
+            shop_exit = create_object(Exit,
+                                      key="front door",
+                                      aliases=["front", "exit", "out", "o"],
+                                      location=shop,
+                                      destination=self.caller.location,
+                                      home=shop)
+            # make a key for accessing the store room
+            storeroom_key_name = "%s-storekey" % shopname
+            storeroom_key = create_object(DefaultObject,
+                                          key=storeroom_key_name,
+                                          location=self.caller,
+                                          home=self.caller)
+            # only allow chars with this key to enter the store room
+            storeroom_entrance.locks.add("traverse:holds(%s)" % storeroom_key_name)
+
+            # inform the builder about progress
+            self.caller.msg("The shop %s was created!" % shop)
+
+class CmdPrice(Command):
+    """
+    Set the price of an item
+
+    Usage:
+        price item value
+
+    This will set the price of an item while inside a storeroom.
+    Value is counted in CC.
+    """
+    key = "price"
+    help_category = "StoreRoom"
+
+    def func(self):
+        try:
+            item_name, value = self.args.split()
+        except:
+            self.caller.msg("I'm afraid I don't understand. Please check your syntax.")
+            return
+        item = self.caller.search(item_name,
+                                  location=self.caller.location,
+                                  typeclass='typeclasses.items.Item',
+                                  quiet=True)
+        if not item:
+            self.caller.msg("I can't find a sellable Item called '{}' here.".format(
+                item_name
+            ))
+            return
+        else:
+            try:
+                value = int(value)
+            except:
+                self.caller.msg("Your price can only be an integer.")
+                return
+            item.db.value = value
+            self.caller.msg("You set the price of {} to {}.".format(
+                item,
+                as_price(value)
+            ))
+
+class StoreRoomCmdSet(CmdSet):
+    "Commands available from within a shop storeroom"
+    def at_cmdset_creation(self):
+        self.add(CmdPrice())
+
+class StoreRoom(Room):
+    "Every shop has a designated storeroom which holds the inventory and provides commands for inventory management"
+
+    def at_object_creation(self):
+        super(StoreRoom, self).at_object_creation()
+        self.cmdset.add_default(StoreRoomCmdSet)
+
+    def return_appearance(self, caller):
+        wares = []
+        for item in self.contents:
+            if item.db.value:
+                wares.append(item.key)
+        desc = super(StoreRoom, self).return_appearance(caller)
+        for line in desc.split("\n"):
+            if line.strip().split("(")[0] in wares:
```

Comment:
I see a slight problem with this approach to highlighting the StoreRoom's stock, which is that the `"("` that you are looking for in the description only appears for the superuser or the objects owner (creator). It may be better to do something like
```:python
    if any(line.startswith(w) for w in wares):
        ...
```
Also, at the top of this module, you have defined a `get_wares()` function that could probably be used in place of lines 260-263.

One last suggestion: it may be nice to append the cost to the item's line in the description in addition to the yellow highlight, since a non superuser/owner shopkeep has the `price` command, but won't have `examine` to be able to check the current price.

Overall, this looks really good. Thanks for the contribution!